h3/src/quic.rs

```diff
@@ -110,7 +112,7 @@ pub trait SendStream<B: Buf> {
     fn poll_ready(&mut self, cx: &mut task::Context<'_>) -> Poll<Result<(), Self::Error>>;
 
     /// Send more data on the stream.
-    fn send_data(&mut self, data: B) -> Result<(), Self::Error>;
+    fn send_data<T: Into<WriteBuf<B>>>(&mut self, data: T) -> Result<(), Self::Error>;
```

Comment:
What does someone have to do to get this `Into<WriteBuf<B>>` to work? Just `impl Buf`, and it works automatically?

Reply:
`WriteBuf` contains an array that will receive the encoded header. While it is most of the time a short slice of bytes, the fact that `VarInts` are used make the maximum possible size of this buffer big enough to slow down method calls that take the `Into<WriteBuf>` by value.  QUIC impls might want to store each data in a collection so it's best to take ownership of them (I guess ?).

Passing `Into<WriteBuf>` allows us to move only the wrapped frame, and allocate the encoding buffer on the stack only when it is necessary. 

But maybe it is possible to do something like `Into<impl Buf>` ?

Reply:
Are you ok with this justification or do you think it's not acceptable / could be a lot better?

Reply:
Yea, the justification seems fine. I was more wondering what does a user of this library write in order to make this work. I didn't examine the implementations closely enough, is it sufficient to just pass anything that implements `Buf`? Or do they need to manually do `impl Into<WriteBuf<MyBuf>> for MyBuf`?

Reply:
Quic implementers have to call `Into<WriteBuf>` when they want to send the frame on the wire. Then they'll get a `Buf` impl. A nice thing here is that they can buffer `Into<WriteBuf>` into a list cheaply, while still being able to get the size of each frame. 

WriteBuf implementation is the role of h3.